src/libexpr/print.cc

```diff
@@ -41,44 +43,55 @@ bool isReservedKeyword(const std::string_view str)
     return reservedKeywords.contains(str);
 }
 
+// Returns 'true' if the character is a symbol that can't be used in a variable name.
+bool isSymbolProhibited(const char & symbol) {
```

Comment:
```suggestion
static bool isSymbolProhibited(const char & symbol) {
```

Or maybe perhaps find another place where this can be used. iirc we have some duplication tech debt around such logic.

Perhaps also a good idea to factor out the negation. In my experience, letting the caller do the negation is a good rule of thumb that prevents double negations when code evolves. ie

```suggestion
bool isValidSymbolChar(const char & symbol) {
```

(and then un-negate the return)

or to make it precise

```suggestion
bool isValidIdentifierTailChar(const char & symbol) {
```

where tail refers to e.g. `builtins.tail`

Reply:
Wow, you're right! I don't know about `isValidIdentifierTailChar` , but `static isValidSymbolChar` definitely looks better! I am going to commit it then.

Reply:
@trofkm the idea with "tail" is we we *also* have duplication for the first character check, so let's factor out *two* such predicate functions.

"head" and "tail" are common way to refer to the first element and rest of the elements of some sort of sequence, so we can use them here.